src/compiler/compile_typescript_code.ts

```diff
@@ -110,17 +144,18 @@ export function bundleFromString(
             crypto: 'crypto-browserify',
             zlib: 'crypto-browserify', // TODO wrong of course
             'internal/deps/acorn/acorn/dist/acorn': `crypto-browserify`, // TODO this is a bug, wasmedge-quickjs should probably add these files
-            'internal/deps/acorn/acorn-walk/dist/walk': `crypto-browserify` // TODO this is a bug, wasmedge-quickjs should probably add these files
+            'internal/deps/acorn/acorn-walk/dist/walk': `crypto-browserify`, // TODO this is a bug, wasmedge-quickjs should probably add these files
+            perf_hooks: path.join(__dirname, 'custom_js_modules/perf_hooks.ts'), // TODO will this work across all operating systems? Might need to test on Mac
+            async_hooks: path.join(
+                __dirname,
+                'custom_js_modules/async_hooks.ts'
+            ),
+            https: path.join(__dirname, 'custom_js_modules/https.ts'),
+            ...esmAliases
         },
-        external: [
-            '_node:fs',
-            '_node:os',
-            '_node:crypto',
-            'qjs:os',
-            '_encoding',
-            'wasi_net',
-            'wasi_http'
-        ]
+        external: [...externalImplemented, ...externalNotImplemented],
+        plugins: [esbuildPluginTsc()]
+        // tsconfig: path.join(__dirname, './esbuild_tsconfig.json')
```

Comment:
What's up with this? Could we either remove it or add a TODO or is it related to the TODO below?

Reply:
I've removed duplication and cleaned up the comments